src/InvestJob.jsx

```diff
@@ -38,32 +39,48 @@ if (process.env.GDAL_DATA) {
   gdalEnv = { GDAL_DATA: process.env.GDAL_DATA }
 }
 
+// TODO: some of these 'global' vars are defined in multiple files
 const CACHE_DIR = 'cache' //  for storing state snapshot files
 const TEMP_DIR = 'tmp'  // for saving datastack json files prior to investExecute
 
+// to translate to the invest CLI's verbosity flag:
+const LOGLEVELMAP = {
+  'DEBUG':   '--debug',
+  'INFO':    '-vvv',
+  'WARNING': '-vv',
+  'ERROR':   '-v',
+}
+
 export class InvestJob extends React.Component {
+  /** This component and it's children render all the visible parts of the app.
+  *
+  * This component's state includes all the data needed to represent one invest
+  * job.
+  */
+
   constructor(props) {
     super(props);
 
     this.state = {
-      sessionID: defaultSessionID(''),
+      sessionID: null,                 // modelName + workspace.directory + workspace.suffix
       modelName: '',                   // as appearing in `invest list`
       modelSpec: {},                   // ARGS_SPEC dict with all keys except ARGS_SPEC.args
       args: null,                      // ARGS_SPEC.args, to hold values on user-interaction
       argsValid: false,                // set on investValidate exit
-      workspace: {                     // only set values when execute completes
-        directory: null, suffix: null},
-      logStdErr: '', 
-      logStdOut: '',
-      sessionProgress: 'home',       // 'home', 'setup', 'log', 'results' (i.e. one of the tabs)
-      activeTab: 'home',
+      workspace: { 
+        directory: null, suffix: null
+      },                               // only set values when execute starts the subprocess
+      logfile: null,                   // path to the invest logfile associated with invest job
+      logStdErr: null,                 // stderr data from the invest subprocess
+      sessionProgress: 'home',         // 'home', 'setup', 'log' - used on loadState to decide which tab to activate
+      jobStatus: null,                 // 'running', 'error', 'success'
+      activeTab: 'home'                // controls which tab is currently visible
     };
     
     this.argsToJsonFile = this.argsToJsonFile.bind(this);

```

Comment:
Hey Dave, could you make a comment about what these bind calls are doing? Not necessarily a code comment, but just a comment here for my curiosity. Do they help in handling callbacks or something?